### PR TITLE
feat(ONYX-620): Activity List: Implement management action sheet

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [ONYX-620] 

### Description

Add management action sheet to the Activity List screen
Hidden behind a feature flag

In this PR I also extract the _mark notifications as read_ functionality into a `markAllNotificationsAsRead()` hook. Having a hook seems to be more convenient.
 
| iOS | Android | Recording |
| --- | --- | --- | 
| simulator_screenshot_75586DCF-3DC2-4E8A-BB35-D5CB19CB205D | Screenshot_1704892101 | <video src="https://github.com/artsy/eigen/assets/36167539/448b11d1-2889-42fd-84e6-d85786a45983"> |

### PR Checklist
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- activity List: Implement management action sheet - daria

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-620]: https://artsyproduct.atlassian.net/browse/ONYX-620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (activity List: Implement management action sheet - daria)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d31e3c42d9a7f5473863d12446a2b7309ad62957
</p>

**Review by olerichter00 — DISMISSED** [review]
Looks great 🌟

File: src/app/Scenes/Activity/Activity.tsx [hunk 1]
```diff
@@ -60,7 +66,33 @@ export const Activity = () => {
...
+                    useModal: true,
+                  },
+                  (buttonIndex) => {
+                    if (buttonIndex === 0) {
+                      markAllNotificationsAsRead()
+                    } else if (buttonIndex === 1) {
+                      navigate("settings/alerts")
+                    } else if (buttonIndex === 2) {
+                      navigate("favorites")
+                    }
```
Comment (MounirDhahri) [inline src/app/Scenes/Activity/Activity.tsx:87]: **minor:** i would suggest using a switch statement here

**Review by MounirDhahri — APPROVED** [review]
Nice! LGTM, thanks for extracting the `useMarkAllNotificationsAsRead`

File: src/app/Scenes/Activity/Activity.tests.tsx [hunk 2]
```diff
@@ -39,8 +39,8 @@ describe("Activity", () => {
 
     await flushPromiseQueue()
 
-    expect(screen.queryByText("Notification One")).toBeTruthy()
-    expect(screen.queryByText("Notification Two")).toBeTruthy()
+    expect(screen.getByText("Notification One")).toBeTruthy()
```
Comment (gkartalis) [inline src/app/Scenes/Activity/Activity.tests.tsx:42]: thank you for addressing those!

**Pull request merged**